hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -278,6 +285,37 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, fmt.Errorf("failed to get cluster %q: %w", req.NamespacedName, err)
 	}
 
+	var res reconcile.Result
+	if r.overwriteReconcile != nil {
```

Comment:
Alternatively to avoid introducing bifurcating paths because of unit tests, this could be a common outer function which receives the reconcile inner function as a parameter. That way the running code has an immutable unique path possible